lib/membrane_framerate_converter.ex

```diff
@@ -0,0 +1,135 @@
+defmodule Membrane.FramerateConverter do
+  @moduledoc """
+  Element converts video to target constant frame rate, by dropping and duplicating frames as necessary
+  Input video may have constant or variable frame rate.
+  Element expects each frame to be received in separate buffer.
+  Additionally, presentation timestamps must be passed in each buffer's metadata.
+  """
+  use Membrane.Filter
+  use Bunch
+
+  alias Membrane.Buffer
+  alias Membrane.Caps.Video.Raw
+
+  require Membrane.Logger
+
+  def_options framerate: [
+                spec: Tuple,
+                default: {30, 1},
+                description: """
+                Target framerate.
+                """
+              ]
+
+  def_output_pad :output,
+    caps: {Raw, aligned: true}
+
+  def_input_pad :input,
+    caps: {Raw, aligned: true},
+    demand_unit: :buffers
+
+  @impl true
+  def handle_init(%__MODULE__{} = options) do
+    state =
+      options
+      |> Map.from_struct()
+      |> Map.merge(%{
+        last_payload: nil,
+        last_metadata: nil,
+        target_pts: 0
+      })
+
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_demand(:output, size, :buffers, _ctx, state) do
+    {{:ok, demand: {:input, size}}, state}
+  end
+
+  @impl true
+  def handle_prepared_to_playing(_ctx, state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_process(
+        :input,
+        %{payload: payload, metadata: metadata} = buffer,
+        _ctx,
+        %{last_payload: nil} = state
+      ) do
+    if not Map.has_key?(metadata, :pts), do: raise("Cannot cut stream without pts")
+
+    state = %{
+      state
+      | target_pts: metadata.pts,
+        last_metadata: metadata,
+        last_payload: payload
+    }
+
+    {{:ok, buffer: {:output, buffer}}, bump_target_pts(state)}
+  end
+
+  @impl true
+  def handle_process(
+        :input,
+        %{payload: payload, metadata: metadata},
+        _ctx,
+        %{last_payload: _last_payload} = state
```

Comment:
I think we can omit pattern matching here as we are not using the variable and we are sure it is always there (because it is set in `handle_init/1`)
```suggestion
       state
```

Reply:
It is marked as resolved but it is neither changed nor commented